db-service/lib/cqn2sql.js

```diff
@@ -1030,50 +1071,92 @@ class CQN2SQLRenderer {
    * @param {Boolean} isUpdate
    * @returns {string[]} Array of SQL expressions for processing input JSON data
    */
-  managed(columns, elements, isUpdate = false) {
-    const annotation = isUpdate ? '@cds.on.update' : '@cds.on.insert'
+  managed(columns, elements) {
+    const cdsOnInsert = '@cds.on.insert'
+    const cdsOnUpdate = '@cds.on.update'
+
     const { _convertInput } = this.class
     // Ensure that missing managed columns are added
     const requiredColumns = !elements
       ? []
-      : Object.keys(elements)
-        .filter(
-          e =>
-            (elements[e]?.[annotation] || (!isUpdate && elements[e]?.default && !elements[e].virtual && !elements[e].isAssociation)) &&
-            !columns.find(c => c.name === e),
-        )
+      : ObjectKeys(elements)
+        .filter(e => {
+          const element = elements[e]
+          // Physical column check
+          if (!element || element.virtual || element.isAssociation) return false
+          // Existance check
+          if (columns.find(c => c.name === e)) return false
+          // Actual mandatory check
+          if (element.default || element[cdsOnInsert] || element[cdsOnUpdate]) return true
```

Comment:
Switch the order of those two checks, property lookup is faster

Reply:
Side effect of trying to make the code readable. Will be adjusted 👍